File: php/features.php
<?php
declare(strict_types=1);

// ShikharMobile SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class ShikharMobileFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new ShikharMobileBaseFeature();
            case "test":
                return new ShikharMobileTestFeature();
            default:
                return new ShikharMobileBaseFeature();
        }
    }
}

File: py/shikharmobile_sdk/features.py
# ShikharMobile SDK feature factory

from shikharmobile_sdk.feature.base_feature import ShikharMobileBaseFeature
from shikharmobile_sdk.feature.test_feature import ShikharMobileTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ShikharMobileBaseFeature(),
        "test": lambda: ShikharMobileTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
